base/bucket_n1ql.go

```diff
@@ -258,10 +258,9 @@ func (bucket *CouchbaseBucketGoCB) WaitForIndexOnline(indexName string) error {
 func (bucket *CouchbaseBucketGoCB) waitForBucketExistence(indexName string, shouldExist bool) error {
 
 	worker := func() (shouldRetry bool, err error, value interface{}) {
-		exists, _, getMetaErr := bucket.GetIndexMeta(indexName)
-		if getMetaErr != nil {
-			return false, getMetaErr, nil
-		}
+		// GetIndexMeta has its own error retry handling,
+		// but keep the retry logic up here for checking if the index exists.
+		exists, _, _ := bucket.GetIndexMeta(indexName)
```

Comment:
If GetIndexMeta returns error, we want to break out of the outer retry loop and return that error.